src/bin/pg_autoctl/cli_config.c

```diff
@@ -252,6 +246,101 @@ cli_config_check_pgsetup(PostgresSetup *pgSetup)
 	{
 		exit(EXIT_CODE_BAD_CONFIG);
 	}
+
+	log_info("Postgres setup for PGDATA \"%s\" is ok, "
+			 "running with PID %d and port %ld",
+			 pgSetup->pgdata, pgSetup->pidFile.port, pgSetup->pidFile.pid);
+}
+
+
+/*
+ * cli_config_check_connections checks that the following three connections are
+ * possible:
+ *
+ *  1. connection to the local Postgres server
+ *  2. connection to the Postgres monitor
+ *  3. streaming replication connection string
+ */
+static void
+cli_config_check_connections(PostgresSetup *pgSetup,
+							 const char *monitor_pguri)
+{
+	PGSQL pgsql = { 0 };
+	char connInfo[MAXCONNINFO] = { 0 };
+
+	pg_setup_get_local_connection_string(pgSetup, connInfo);
+	pgsql_init(&pgsql, connInfo, PGSQL_CONN_LOCAL);
+
+	if (!pgsql_is_in_recovery(&pgsql, &pgSetup->is_in_recovery))
+	{
+		/* errors have already been logged */
+		exit(EXIT_CODE_PGSQL);
+	}
+
+	log_info("Connection to local Postgres ok, using \"%s\"", connInfo);
+
+	/*
+	 * Do not check settings on the monitor node itself.
+	 */
+	if (monitor_pguri != NULL)
```

Comment:
we are checking this twice and never using it if it is null.

can we have an early exit if monitor_pguri is null ?

Reply:
Fixed, thanks. I though about that code as two separate units, but you're right that it makes sense to implement an early return, it's now easier to read.